api/firmware/device.go

```diff
@@ -241,11 +249,25 @@ func (device *Device) Init() error {
 	device.channelHashDeviceVerified = false
 	device.sendCipher = nil
 	device.receiveCipher = nil
-	device.changeStatus(StatusConnected)
 
-	if err := device.inferVersionAndProduct(); err != nil {
-		return err
+	version, product, unlocked, initialized, err := device.info()
+	if err != nil {
+			return errp.New(
+				"OP_INFO unavailable; need to provide version and product via the USB HID descriptor")
+	}
+	// The semantics of the firmware and status.Status are not the same.
+	// status.Status StatusInitialized means the device is unlocked.
+	// The firmware initialized = true means the device can be unlocked.
+	if unlocked {
+		device.changeStatus(StatusInitialized)
```

Comment:
I guess we could also `return` here so we don't have to do this:
```
if (device.version.AtLeast(semver.NewSemVer(2, 0, 0)) && (device.status != StatusInitialized)) {
```
later? I was not sure if there is other stuff that needs to be performed (e..g handshake) but if the device is initialized and unlocked there should be nothing else to do right ?

Reply:
edit: **no** I tested again and pairing for example might be performed even if the device is initialized.

(I unlocked the device on `AppImage` and kept it connected, closed the `AppImage` and started `webdev` the pairing popped up, but afterwards the device was still unlocked.)